tests/test_record_samples.rs

```diff
@@ -169,6 +169,60 @@ fn test_event_xml_sample_with_user_data() {
     );
 }
 
+#[test]
+fn test_event_xml_sample_with_entity_ref() {
```

Comment:
Also generate and test both samples for JSON

Reply:
done.